### PR TITLE
Add blacklist whitelist for headers

### PR DESCRIPTION
**Pull request by ger-benjamin**
For GEO-4639

Add blacklist/whitelist system for headers.
Add docs.

@sbrunner Are you sure you don't want to add this feature per url like the global_header system ? 

```
vars:
  global_headers:
    - pattern: <regex>
      headers:
        <header>: <value>
```
It looks risky, or possibly problematic to apply these lists on all request going through the proxy.
(@jwkaltz it's for Agrola, but I add you especially to review the documentation)

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/views/proxy.py [hunk 1]
```diff
@@ -100,11 +103,24 @@ def _proxy(self, url, params=None, method=None, cache=False, body=None, headers=
...
 
         if method in ("POST", "PUT") and body is None:  # pragma: no cover
             body = self.request.body
 
+        # Filters headers with a whitelist.
+        # Some default pyramid headers will be added back by pyramid.
+        if len(self.headers_whitelist) > 0:
+            for header in list(headers):
+                if header not in self.headers_whitelist:
+                    headers.pop(header)
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/views/proxy.py:116]:
```suggestion
            headers = {key: value for key, value in headers.items() if key in self.headers_whitelist}
```

File: geoportal/c2cgeoportal_geoportal/views/proxy.py [hunk 2]
```diff
@@ -100,11 +103,24 @@ def _proxy(self, url, params=None, method=None, cache=False, body=None, headers=
...
+
+        # Filters headers with a blacklist.
+        # Some default pyramid headers will be added back by pyramid.
+        for header in list(headers):
+            if header in self.headers_blacklist:
+                headers.pop(header)
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/views/proxy.py:122]:
```suggestion
        headers = {key: value for key, value in headers.items() if key not in self.headers_blacklist}
```

**Review by sbrunner — COMMENTED** [review]
Looks good, just propose use functional syntax

**Comment by sbrunner** [comment]
> Are you sure you don't want to add this feature per url like the global_header system?

I don't think that necessary...

**Comment by ger-benjamin** [comment]
> The real error is here...

It's fixed in my 2nd commit. I don't know why I've that. The error append also on branch `2.5`

**Review by sbrunner — APPROVED** [review]
Looks good :-)

**Pull request merged**